ledger/internal/evalprefetcher.go

```diff
@@ -508,7 +508,17 @@ func (p *accountPrefetcher) asyncPrefetchRoutine(queue *preloaderTaskQueue, task
 			task.address = &creator
 		}
 		var resource ledgercore.AccountResource
-		resource, err = p.ledger.LookupResource(p.rnd, *task.address, task.creatableIndex, task.creatableType)
+		if task.creatableType == basics.AppCreatable {
+			var appResource ledgercore.AppResource
+			appResource, err = p.ledger.LookupApplication(p.rnd, *task.address, basics.AppIndex(task.creatableIndex))
+			resource.AppParams = appResource.AppParams
```

Comment:
It looks that with your change, we should also change the `loadedTransactionGroup.resources` into the two explicit lists of assets and applications. This could be done later on, though.

Reply:
yes, I wasn't sure if that was in scope or whether it was OK for behind-the-scenes implementations further away from the external-facing ledger interfaces to deal with resources. But yes it would mean splitting `loadAccountsAddResourceTask` into two functions/tasks and maybe could be put off for when someone works on the harness to catch differences between calls to apply.Balances from test transactions and the resources that are queued for prefetching.